File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.cpp
class Solution {
public:
 static bool cmp(pair<int,int>&a,pair<int,int>&b){
  return a.second > b.second;
}
    vector<int> topKFrequent(vector<int>& nums, int k) {
        map<int,int>mp;
        vector<int>ans;
    
        for(int i=0;i<nums.size();i++){
          mp[nums[i]]++;
          
        }
        vector<pair<int,int>>freq(mp.begin(),mp.end());
        sort(freq.begin(),freq.end(),cmp);
       
       for(auto it :freq ){
         if(k == 0) return ans;
          ans.push_back(it.first);
          k--;
      
       
        }

        return ans;
    }
};

File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.java
class Solution {

    class Pair{
        int elem;
        int freq;

        Pair(int elem, int freq){
          this.elem = elem;
          this.freq = freq;
        }
    }
    public int[] topKFrequent(int[] nums, int k) {
       
        Map<Integer, Integer>mp = new HashMap<>();
        PriorityQueue<Pair> pq = new PriorityQueue<>((a,b)-> b.freq - a.freq);
        int[] ans = new int[k];
        for(int i=0;i<nums.length;i++){
           mp.put(nums[i], mp.getOrDefault(nums[i], 0) + 1);
        }
        for(Map.Entry<Integer, Integer> m : mp.entrySet()){
           pq.add(new Pair(m.getKey(), m.getValue()));
        }
        for(int i=0;i<k;i++){
          ans[i] = pq.poll().elem;
        }
      return ans;
    }
}